## 50_joc-toggle-buffer.diff by Peter S Galbraith <psg@debian.org>

--- a/elisp/emacs-goodies-el/joc-toggle-buffer.el	2004-04-20 14:44:06.000000000 -0400
+++ b/elisp/emacs-goodies-el/joc-toggle-buffer.el	2004-04-20 14:46:48.000000000 -0400
@@ -1,4 +1,4 @@
-;;; @(#) toggle-buffer.el --- flips back and forth between two buffers
+;;; @(#) joc-toggle-buffer.el --- flips back and forth between two buffers
 
 ;; Copyright (C) 2001 by Joseph L. Casadonte Jr.
 
@@ -62,7 +62,7 @@
 ;;  Put this file on your Emacs-Lisp load path and add the following to your
 ;;  ~/.emacs startup file
 ;;
-;;     (require 'toggle-buffer)
+;;     (require 'joc-toggle-buffer)
 
 ;;; Usage:
 ;;
@@ -99,13 +99,18 @@
 ;;  Any comments, suggestions, bug reports or upgrade requests are welcome.
 ;;  Please send them to Joe Casadonte (emacs@northbound-train.com).
 ;;
-;;  This version of toggle-buffer was developed and tested with NTEmacs 20.5.1
+;;  This version of joc-toggle-buffer was developed and tested with NTEmacs 20.5.1
 ;;  and 2.7 under Windows NT 4.0 SP6 and Emacs 20.7.1 under Linux (RH7).
 ;;  Please, let me know if it works with other OS and versions of Emacs.
 
 ;;; Change Log:
 ;;
 ;;  see http://www.northbound-train.com/emacs/toggle-buffer.log
+;;
+;;  2003-11-23 Peter S Galbraith <psg@debian.org>
+;;   This version, distributed in the Debian package `emacs-goodies-el',
+;;   was renamed from toggle-buffer.el to joc-toggle-buffer.el.  The prefix
+;;   was also added to a few variables.
 
 ;;; **************************************************************************
 ;;; **************************************************************************
@@ -151,7 +156,7 @@
   :group 'joc-toggle-buffer)
 
 ;; ---------------------------------------------------------------------------
-(defcustom toggle-buffer-load-hook nil
+(defcustom joc-toggle-buffer-load-hook nil
   "Hook to run when package is loaded."
   :type 'hook
   :group 'joc-toggle-buffer)
@@ -181,10 +186,12 @@
 (defvar joc-toggle-buffer-last-buffer nil
   "Contains the name of the previous buffer.")
 
+;;;###autoload
 (defun joc-toggle-buffer ()
   "Switch to previous active buffer."
   (interactive)
-  (if (not (boundp 'joc-toggle-buffer-last-buffer))
+  (if (or (not (boundp 'joc-toggle-buffer-last-buffer))
+          (not joc-toggle-buffer-last-buffer))
 	  (error "No previous buffer to switch to (yet)"))
   (let ((buff (get-buffer joc-toggle-buffer-last-buffer)))
 	(if (not buff)
@@ -231,9 +238,8 @@
 ;;; **************************************************************************
 ;;; ***** we're done
 ;;; **************************************************************************
-(provide 'toggle-buffer)
-(run-hooks 'toggle-buffer-load-hook)
+(run-hooks 'joc-toggle-buffer-load-hook)
+
+(provide 'joc-toggle-buffer)
 
 ;;; toggle-buffer.el ends here
-;;; **************************************************************************
-;;;; *****  EOF  *****  EOF  *****  EOF  *****  EOF  *****  EOF  *************
